jni/src/test/java/com/horizen/commitmenttreenative/ScCommitmentCertPathTest.java

```diff
@@ -0,0 +1,159 @@
+                ScCommitmentCertPath path = getCertPath(ct, scId, certLeaf);
+                FieldElement fakeRoot = FieldElement.createRandom();) {
+            assertFalse(path.verify(fakeRoot, scId, certLeaf));
+        }
+    }
+
+    @Test
+    public void verifyShouldFailIfInvalidScId() {
+        try (
+                CommitmentTree ct = CommitmentTree.init();
+                FieldElement scId = FieldElement.createRandom();
+                FieldElement certLeaf = FieldElement.createRandom();
+                ScCommitmentCertPath path = getCertPath(ct, scId, certLeaf);
+                FieldElement root = ct.getCommitment().get();
+                FieldElement fakeScId = FieldElement.createRandom();) {
+            assertFalse(path.verify(root, fakeScId, certLeaf));
+        }
+    }
+
+    @Test
+    public void verifyShouldFailIfInvalidCertHash() {
+        try (
+                CommitmentTree ct = CommitmentTree.init();
+                FieldElement scId = FieldElement.createRandom();
+                FieldElement certLeaf = FieldElement.createRandom();
+                ScCommitmentCertPath path = getCertPath(ct, scId, certLeaf);
+                FieldElement root = ct.getCommitment().get();
+                FieldElement fakeCertHash = FieldElement.createRandom();) {
+            assertFalse(path.verify(root, fakeCertHash, certLeaf));
+        }
+    }
+
+    @Test
+    public void generateScTxCommitmentRoot() {
+        try (
+                CommitmentTree ct = CommitmentTree.init();
+                FieldElement scId = FieldElement.createRandom();
+                FieldElement certLeaf = FieldElement.createRandom();
+                ScCommitmentCertPath path = getCertPath(ct, scId, certLeaf);
+                FieldElement root = ct.getCommitment().get();) {
+            assertEquals(root, path.apply(scId, certLeaf).get());
+        }
+    }
+
+    @Test
+    public void shouldSerializeDeserialize() {
+        try (
+                CommitmentTree ct = CommitmentTree.init();
+                FieldElement scId = FieldElement.createRandom();
+                FieldElement certLeaf = FieldElement.createRandom();
+                ScCommitmentCertPath path = getCertPath(ct, scId, certLeaf);
+                FieldElement root = ct.getCommitment().get();) {
+            /// Sanity Check
+            assertTrue(path.verify(root, scId, certLeaf));
+
+            byte[] serialized = path.serialize();
+
+            try (ScCommitmentCertPath deserialized = ScCommitmentCertPath.deserialize(serialized, true)) {
+                assertTrue(deserialized.verify(root, scId, certLeaf));
+            }
+
+            serialized[0] = (byte) (serialized[0] + 1);
+
+            try (ScCommitmentCertPath deserialized = ScCommitmentCertPath.deserialize(serialized, true)) {
+                assertFalse(deserialized.verify(root, scId, certLeaf));
+            }
+        }
+    }
+
+    @Test
+    public void completeTest() {
```

Comment:
could be nice to  test also it is working correctly with mutliple sidechains in the same tree

Reply:
Should be already covered on the rust side. But if you prefer I can add another one here.

Reply:
ok is enough